Add application block (#26474)
idaholab/moose: framework/doc/content/source/actions/CreateApplicationBlockAction.md

# CreateApplicationBlockAction

!syntax description /Application/CreateApplicationBlockAction

> This action is used to add `Application` block as a MOOSE object.

Reviewer: ```suggestion
This action is parse the parameters passed to the `[Application]` block.
```

I dont think it s a moose object?

> This action is parse the parameters passed to the `[Application]` block.

Reviewer: `parses`

but I thought we were not doing this through an action in the end

Author: I think for this PR  we decide to use action, right  @loganharbour?  I'll do another PR to create a application system and maybe parse the parameters inside Appfactory

Author: It's there

> More information about `Application` block can be found on the [Application syntax page](syntax/Application/index.md).

Reviewer: ```suggestion
More information about the `Application` block can be found on the [Application syntax page](syntax/Application/index.md).
```
